### DISCUSSION TITLE
Beim Bauen von master kann crti.o, libc und crtn.o nicht gefunden werden

### DISCUSSION THREAD

**Discussion by SvenLuebke** [Freetz-NG/freetz-ng#327] [Allgemein]
Hallo @fda77!

Hast Du in letzter Zeit mal ein distclean gemacht? Ich kann auf einem Debian10 derzeit nicht mehr durchbauen...kann aber gut sein, dass der Fehler bei mir liegt. Diesen Fehler gibt es seit max. 24h:

```
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -Wall -O2 -m32 -MT libfakeroot.lo -MD -MP -MF .deps/libfakeroot.Tpo -c ../../libfakeroot.c  -fPIC -DPIC -o .libs/libfakeroot.o

In file included from /usr/include/bits/errno.h:26,
                 from /usr/include/errno.h:28,
                 from ../../communicate.c:48:
/usr/include/linux/errno.h:1:10: fatal error: asm/errno.h: No such file or directory
 #include <asm/errno.h>
          ^~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:666: communicate.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
In file included from /usr/include/bits/errno.h:26,
                 from /usr/include/errno.h:28,
                 from ../../libfakeroot.c:145:
/usr/include/linux/errno.h:1:10: fatal error: asm/errno.h: No such file or directory
 #include <asm/errno.h>
          ^~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:666: libfakeroot.lo] Error 1
make[1]: Leaving directory '/media/projects/fritzbox/freetz/freetz-ng/source/host-tools/fakeroot-1.25.3/build/biarch'
make: *** [tools/make/fakeroot-host/fakeroot-host.mk:74: /media/projects/fritzbox/freetz/freetz-ng/tools/build/lib32/libfakeroot-0.so] Error 2
```

Diesen Fehler hatte ich vorher:

```
/media/projects/fritzbox/freetz/freetz-ng/source/toolchain-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/gcc-5.5.0-final/./gcc/xgcc -B/media/projects/fritzbox/freetz/freetz-ng/source/toolchain-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/gcc-5.5.0-final/./gcc/ -B/media/projects/fritzbox/freetz/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/mips-linux-uclibc/bin/ -B/media/projects/fritzbox/freetz/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/mips-linux-uclibc/lib/ -isystem /media/projects/fritzbox/freetz/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/mips-linux-uclibc/include -isystem /media/projects/fritzbox/freetz/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/mips-linux-uclibc/sys-include    -O2  -march=34kc -mtune=34kc -msoft-float -O3 -Wa,--trap -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wl,-I/var/media/ftp/usr/lib/ld-uClibc.so.1 -minterlink-mips16 -DIN_GCC  -DCROSS_DIRECTORY_STRUCTURE  -W -Wall -Wno-narrowing -Wwrite-strings -Wcast-qual -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition  -isystem ./include   -fPIC -g -DIN_LIBGCC2 -fbuilding-libgcc -fno-stack-protector  -shared -nodefaultlibs -Wl,--soname=libgcc_s.so.1 -Wl,--version-script=libgcc.map -o ./libgcc_s.so.1.tmp -march=34kc -mtune=34kc -msoft-float -O3 -Wa,--trap -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -Wl,-I/var/media/ftp/usr/lib/ld-uClibc.so.1 -minterlink-mips16 -B./ _muldi3_s.o _negdi2_s.o _lshrdi3_s.o _ashldi3_s.o _ashrdi3_s.o _cmpdi2_s.o _ucmpdi2_s.o _clear_cache_s.o _trampoline_s.o __main_s.o _ctors_s.o _ffsdi2_s.o _clzdi2_s.o _ctzsi2_s.o _ctzdi2_s.o _popcount_tab_s.o _popcountsi2_s.o _popcountdi2_s.o _paritysi2_s.o _paritydi2_s.o _powisf2_s.o _powidf2_s.o _powixf2_s.o _powitf2_s.o _mulsc3_s.o _bswapsi2_s.o _bswapdi2_s.o _clrsbsi2_s.o _clrsbdi2_s.o _fixunsxfsi_s.o _fixxfdi_s.o _fixtfdi_s.o _fixunsxfdi_s.o _fixunstfdi_s.o _floatdixf_s.o _floatditf_s.o _floatundixf_s.o _floatunditf_s.o _fixsfti_s.o _fixdfti_s.o _fixxfti_s.o _fixtfti_s.o _fixunssfti_s.o _fixunsdfti_s.o _fixunsxfti_s.o _fixunstfti_s.o _floattisf_s.o _floattidf_s.o _floattixf_s.o _floattitf_s.o _floatuntisf_s.o _floatuntidf_s.o _floatuntixf_s.o _floatuntitf_s.o _divdi3_s.o _moddi3_s.o _udivdi3_s.o _umoddi3_s.o _udiv_w_sdiv_s.o _udivmoddi4_s.o addsf3_s.o divsf3_s.o eqsf2_s.o gesf2_s.o lesf2_s.o mulsf3_s.o negsf2_s.o subsf3_s.o unordsf2_s.o fixsfsi_s.o fixunssfsi_s.o floatsisf_s.o floatunsisf_s.o fixsfdi_s.o fixunssfdi_s.o floatdisf_s.o floatundisf_s.o adddf3_s.o divdf3_s.o eqdf2_s.o gedf2_s.o ledf2_s.o muldf3_s.o negdf2_s.o subdf3_s.o unorddf2_s.o fixdfsi_s.o fixunsdfsi_s.o floatsidf_s.o floatunsidf_s.o fixdfdi_s.o fixunsdfdi_s.o floatdidf_s.o floatundidf_s.o extendsfdf2_s.o truncdfsf2_s.o enable-execute-stack_s.o unwind-dw2_s.o unwind-dw2-fde-dip_s.o unwind-sjlj_s.o unwind-c_s.o emutls_s.o libgcc.a -lc && rm -f ./libgcc_s.so && if [ -f ./libgcc_s.so.1 ]; then mv -f ./libgcc_s.so.1 ./libgcc_s.so.1.backup; else true; fi && mv ./libgcc_s.so.1.tmp ./libgcc_s.so.1 && (echo "/* GNU ld script"; echo "   Use the shared library, but some functions are only in"; echo "   the static library.  */"; echo "GROUP ( libgcc_s.so.1 -lgcc )" ) > ./libgcc_s.so
/media/projects/fritzbox/freetz/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/mips-linux-uclibc/bin/ld: cannot find crti.o: No such file or directory
/media/projects/fritzbox/freetz/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/mips-linux-uclibc/bin/ld: cannot find -lc
/media/projects/fritzbox/freetz/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/mips-linux-uclibc/bin/ld: cannot find crtn.o: No such file or directory
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:953: libgcc_s.so] Error 1
make[3]: Leaving directory '/media/projects/fritzbox/freetz/freetz-ng/source/toolchain-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/gcc-5.5.0-final/mips-linux-uclibc/libgcc'
make[2]: *** [Makefile:11331: all-target-libgcc] Error 2
make[2]: Leaving directory '/media/projects/fritzbox/freetz/freetz-ng/source/toolchain-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/gcc-5.5.0-final'
make[1]: *** [Makefile:873: all] Error 2
make[1]: Leaving directory '/media/projects/fritzbox/freetz/freetz-ng/source/toolchain-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/gcc-5.5.0-final'
make: *** [toolchain/make/target/gcc/gcc.mk:205: /media/projects/fritzbox/freetz/freetz-ng/source/toolchain-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/gcc-5.5.0-final/.compiled] Error 2
```

Hat jemand eine Idee?

Danke Euch und viele Grüße
Sven

**Comment by fda77** [comment]
.config zum testen anhängen...
Als workaround könntest du von fakeroot auf pseudo im menuconfig umstellen
https://stackoverflow.com/questions/14795608/asm-errno-h-no-such-file-or-directory

**Reply by SvenLuebke** [reply]
Also mindestens der erste Fehler liegt nicht an neuen freetz-ng commits. Das hab ich heute hervorgerufen, weil ich etwas für aarch64 cross compilen musste. Das habe ich zu dem Thema gefunden:
https://www.mail-archive.com/debian-gcc@lists.debian.org/msg57728.html
Ich hab etwas gebraucht...aber der Fehler kam mir bekannt vor. Die Lösung, die man mit google findet, sieht so aus, dass man einen symbolischen Link anlegen muss. Das hab ich getan. Danach kam der nächste Fehler (Stichwort: cap_user_header_t "error: unknown type name"). Da erinnerte ich mich wieder, dass ich das alles schon mal irgendwie gelöst hatte. Ich hab nun alle Patches von fakeroot-host bis auf "100-no_doc_test.patch" deaktiviert. Danach kompiliert es.

Gepackte .config (ist eigentlich ein 7zip) angehängt
[config.zip](https://github.com/Freetz-NG/freetz-ng/files/6640728/config.zip)
Die config wird wohl ein bisschen Ärger machen, oder? Allein schon durch FREETZ_PACKAGE_SUBVERSION_VERSION_1_14

Jetzt bin ich bei zuletzt genanntem Fehler angekommen. Dieser entsteht durch meine Konfiguration von FREETZ_RPATH. Der Pfad wird zur uclibc übertragen (im menuconfig der uclibc unter "library path component": (var/media/ftp/usr/lib) ), und die als fehlend gemeldete Datei liegt dann unter

toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/var/media/ftp/usr/lib/crti.o

Hier fehlt wahrscheinlich das Setzen des Linker-Parameters -L an die richtige Stelle.

**Comment by fda77** [comment]
Ich blick nicht so ganz durch. Setzt doch am besten zuerst mal einen Titel für die Diskussion. 
Was hat irgend ein andere Comiliercorgang mit Freetz zu tun? Hast du dafür was am System verändert? Wer stellt denn crti.o und kopiert di falsch? Wenn du eine x64 CPU hast, warum bast du für x86 
Wenn du ne neuere SVN version nutzen willst, lad die doch als PR oder Patch hoch

**Reply by SvenLuebke** [reply]
> Ich blick nicht so ganz durch. Setzt doch am besten zuerst mal einen Titel für die Diskussion.

Ja, Du hast Recht, ich werde den Titel ändern. Mit der angefangenen Diskussion wollte ich erstmal nur prüfen, ob ich der Einzige bin, der nicht mehr bauen kann. Das sieht ja derzeit danach aus und hilft mir weiter!

> Was hat irgend ein andere Comiliercorgang mit Freetz zu tun? Hast du dafür was am System verändert?

Das frage ich mich auch. Es sieht so aus, als würde die Installation des gcc-cross-compilers (für die aarch64-Architektur) dazu führen, dass man andere Architekturen nicht mehr bauen kann. Woran das liegt, kann ich nicht sagen. Ob ich vor ein paar Monaten mal etwas am System verändert habe, kann ich nicht mehr sagen, außer dass ich auf der Suche nach dem 1. Fehler den oben erwähnten symbolischen Link erzeugt habe mittels
`ln -s /usr/include/asm-generic/ /usr/include/asm`
Damit ist zumindest der erste Fehler verschwunden. Ich werde sicherheitshalber mal eine neue Debian-Installation ausprobieren. 

>Wer stellt denn crti.o und kopiert di falsch? Wenn du eine x64 CPU hast, warum bast du für x86

Laut der Meldung ist das die crti.o für die libgcc. Warum der Linker die nicht findet, habe ich bisher noch nicht herausgefunden. Woran erkennst Du, dass ich für x86 baue? Ich habe ein x64-System.

> Wenn du ne neuere SVN version nutzen willst, lad die doch als PR oder Patch hoch

Ja, das wollte ich schon immer mal machen, bin aber noch nicht dazu gekommen. Zu viele Sachen parallel und zu wenig Zeit.

**Reply by SvenLuebke** [reply]
Hi!

Aktueller Zwischenstand:
Ich bin jetzt endlich mal dazu gekommen Debian 10 neu zu installieren. Ohne "System-eigene" aarch64-Architektur kann ich auf dem neuen System scheinbar ohne Probleme durchbauen. Nach Installation des gcc-8-aarch64* wird - mehr oder weniger automatisch - die gcc-multilib entfernt. Wenn man dann "make" startet, bekommt man diverse Fehlermeldungen. Nach Reinstallation der multilib kann ich auf dem neuen System ohne Probleme durchbauen. Irgendetwas hatte ich auf meinem alten System wohl scheinbar kaputtgebastelt.

Mir ist allerdings bei beiden Buildvorgängen (und aus der Erinnerung auch schon bei meinem alten Debian-System) ein Fehler beim Kompilieren von busybox aufgefallen:

```
  CC      util-linux/hexdump_xxd.o
  CC      util-linux/mkswap.o
  LD      util-linux/volume_id/built-in.o
  CC      util-linux/volume_id/exfat.o
  CC      util-linux/more.o
  CC      util-linux/volume_id/f2fs.o
  CC      util-linux/mount.o
  CC      util-linux/volume_id/get_devname.o
  AR      networking/lib.a
  CC      util-linux/volume_id/util.o
  DOC     BusyBox.html
  CC      util-linux/volume_id/volume_id.o
  CC      util-linux/pivot_root.o
  AR      util-linux/volume_id/lib.a
  CC      util-linux/renice.o
  CC      util-linux/swaponoff.o
  CC      util-linux/switch_root.o
  CC      shell/math.o
  CC      shell/random.o
  CC      util-linux/taskset.o
  CC      shell/shell_common.o
  CC      util-linux/umount.o
  AR      util-linux/lib.a
  AR      shell/lib.a
  LINK    busybox_unstripped
Trying libraries: rt m crypt
 Library rt is not needed, excluding it
 Library m is not needed, excluding it
 Library crypt is not needed, excluding it
Final link with: <none>
make[1]: Leaving directory '/media/projects/fritzbox/freetz-ng_aarch64/source/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox-1.33.1'
mkdir -p packages/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox/; cp source/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox-1.33.1/busybox packages/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox/busybox; [ "y" == "y" ] && tools/patchelf --set-interpreter /var/media/ftp/usr/lib/ld-uClibc.so.1 packages/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox/busybox || true; /media/projects/fritzbox/freetz-ng_aarch64/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/bin/mips-linux-uclibc-strip --remove-section={.comment,.note,.pdr} packages/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox/busybox;
/media/projects/fritzbox/freetz-ng_aarch64/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/bin/mips-linux-uclibc-strip: packages/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox/steWMmhm: Not enough room for program headers, try linking with -N
/media/projects/fritzbox/freetz-ng_aarch64/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/mips-linux-uclibc/bin/mips-linux-uclibc-strip:packages/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox/steWMmhm[.MIPS.abiflags]: Bad value
make: *** [make/busybox/busybox.mk:60: packages/target-mips_gcc-5.5.0_uClibc-1.0.37-nptl_kernel-3.10/busybox/busybox] Error 1
user@debian10test:/media/projects/fritzbox/freetz-ng_aarch64$
```
@fda77: Kannst Du diesen vielleicht reproduzieren, kennst einen Grund oder eine Lösung?

Nachdem man make erneut ausführt ist der Fehler dann weg.

**Reply by fda77** [reply]
`Not enough room for program headers, try linking with -N` wurde kürzlich hier zu "sudo" gemeldet: https://github.com/Freetz-NG/freetz-ng/discussions/339
Vielleicht wird das durch ein Update am Hostsystem verursacht? Ich hatte das bislang noch nie

---

Manche (wenige) Dinge von Freetz erforder 32-bit bzw multiarch. Spontan fällt mir nur dtc-host von @PeterPawn ein: https://github.com/Freetz-NG/freetz-ng/blob/master/tools/make/dtc-host/dtc-host.mk#L25 
Er wirft gerne mit Links zu https://de.wikipedia.org/wiki/Unix-Philosophie um sich, überliest selbst aber die Punkte mit der "Portierbarkeit"...
MIT einem aarch64 System kann man somit kein Freetz compilieren, da damit nur x64 compilierbar ist, im Gegensatz zu x86_64.

**Reply by PeterPawn** [reply]
> Spontan fällt mir nur dtc-host von @PeterPawn ein

Schwach, ganz schwach ... was genau habe ich denn mit dem Kompilieren von `dtc-host` zu tun? Der gesamte Inhalt dieses Verzeichnisses stammt - von vorne bis hinten - von @er13.

Vielleicht solltest Du - anstatt sinnlos herumzugeifern - einfach mal richtig nachdenken, BEVOR Du solchen Unsinn schreibst? 

Vermutlich kannst Du einfach mal wieder `yourfritz-akc-host` und `dtc-host` nicht auseinanderhalten? Und hast obendrein **schon wieder** vergessen (langsam kristallisiert sich das als klinisches Symptom heraus), daß auch das in Freetz (und damit in Freetz-NG) verwendete `yourfritz-akc-host` (https://github.com/Freetz-NG/freetz-ng/tree/master/tools/make/yourfritz-akc-host) nur noch von der Grundidee her auf meinen Quellen basiert?

> überliest selbst aber die Punkte mit der "Portierbarkeit"

Was genau versprichst Du Dir denn davon, solchen hanebüchenen Unsinn zu verbreiten, den jeder, der selbst etwas lesen kann, umgehend durchschauen sollte? Bist Du darauf aus, daß man Dein Projekt künftig komplett ignoriert und selbst seltenes, spontanes Lesen der "Neuigkeiten" oder von Fragen/Problemen der Benutzer einstellt, damit Du - "im eigenen Saft schmorend" - vor Dich hinwursteln kannst?

Ich weiß nicht mehr genau, WIE OFT ich nun schon versucht habe Dir zu erklären, daß ein Einsatz von `yourfritz-akc-host` nur dann überhaupt sinnvoll und erforderlich ist, wenn der Benutzer tatsächlich den AVM-Kernel durch einen eigenen ersetzen will. Ohne diese Absicht, braucht es das ganze Programm nicht. 

Und zweitens arbeitet nun mal der AVM-Kernel auch nur mit 32-Bit - also sind die dort enthaltenen Pointer logischerweise auch 32-Bit-Pointer, was sowohl mein Original als auch der von @er13 abgewandelte Code (und das ist der, den auch Freetz-NG benutzt) beherrschen ... aber NIEMAND kann (und wird) es Dir verbieten, die bisher in `yourfritz-akc-host` (oder auch in meinem originalen Code) verwendete 32-Bit-Arithmetik im Umgang mit solchen Pointern auf 64-Bit aufzubohren. 

Nur ist das eben nicht einfach so im Handumdrehen gemacht (erst recht nicht dann, wenn der Source-Code sowohl auf 32- als auch auf 64-Bit-Systemen parallel funktionieren soll) und bei den paar Modellen, wo bisher in Freetz-NG tatsächlich ein "replace kernel" möglich wäre (m.W. hat immer noch niemand das korrekte Packen für GRX-Boxen implementiert - obwohl die Eckdaten bekannt sind und das ja nun wirklich zu den simpleren Aufgabenstellungen gehört), lohnt sich das vermutlich auch gar nicht wirklich.

Pack den ganzen Teil mit `yourfritz-akc-host` und `dtc-host` einfach an die Stelle, wo er hingehört - nämlich zum Linken eines eigenen Kernels. Ohne dieses Linken wird keines der beiden Projekte jemals benötigt ... und obendrein muß der jeweils aus dem AVM-Kernel extrahierte Inhalt auch noch exakt zum Modell und zur FRITZ!OS-Version passen (damit die Speicherangaben für die AVM-Module korrekt sind). 

Was hat der ganze Zinnober dann an der Stelle zu suchen, wo die "Umgebung" (Compiler, C-Library, Host-Tools) erstellt wird, die ja durchaus über mehrere Modelle hinweg identisch ist? Richtig ... gar nichts - auch wenn Du das offenbar nicht verstehen willst (mittlerweile beschleichen mich Zweifel, ob Du es überhaupt kannst).

Wenn das richtig geändert wurde, dann kann man auch auf "exotischen" 64-Bit-Systemen (wie AARCH64 - also 64-bittigen ARM-Systemen) problemlos Freetz-NG benutzen (oder zumindest sollte man das tun können, wenn der Maintainer "alles richtig" macht) ... lediglich das (ohnehin immer seltener unterstützte) "replace kernel" bleibt einem dann verwehrt. 

Wenn Du das irgendwann mal soweit umgesetzt haben solltest, daß die `make`-Aufrufe für diese zwei Projekte tatsächlich an der richtigen Stelle erfolgen, dann denke ich vielleicht auch noch einmal darüber nach, das Programm (meinerseits) auch auf 64-Bit lauffähig zu machen (obwohl mich persönlich weder Freetz noch Freetz-NG weiterhin interessiert) - aber nicht als "Abkürzung" für Dich, damit Du Dir irgendwelche Arbeiten sparen kannst. @er13 hat das damals halt anders implementiert und schon da hatte ich meinerseits Einwände dagegen - kannst Du jederzeit im IPPF nachlesen. 

Aber @er13 hatte dann wenigsten noch den Anstand, seine eigenen Vorstellungen auch selbst umzusetzen und nicht ständig anderen auf den Geist zu gehen, sie sollten das und das irgendwie so ändern, daß es für ein "fremdes" Projekt (und das ist `Freetz-NG` für mich weiterhin) paßt, bei dem sie sich eigentlich auch nur widerwillig engagieren und wenn, dann am ehesten, um denen Hilfestellungen zu bieten, die ansonsten von den Maintainern (auch wenn Du wohl immer noch "einzeln" bist, zeugen ja auch diverse Synonyme von einem eher schwierigen Verständnis der eigenen Persönlichkeit, so daß hier vielleicht der Plural auch gerechtfertigt ist?) im Stich gelassen werden - entweder wegen irgendwelcher kindischer Zwistigkeiten, die man dann gerne hier in aller Öffentlichkeit "auslebt" oder manchmal auch schlicht mangels Expertise. DABEI bin ich dann bereit zu helfen (oder war es bisher zumindest, wie einige Beispiele zeigen) - aber nicht länger dann, wenn es ständig irgendwelche (unbegründeten) "Seitenhiebe" gibt, wie den oben verlinkten.

Aber noch einmal zurück zu der Frage, wer denn nun tatsächlich für den ganzen `avm_kernel_config`-Part zuständig ist ... denn das ist eben ein weiterer Grund, warum Du mir nicht länger mit dem Gegreine auf den Sack gehen solltest (vielleicht kapierst Du es ja, wenn ich etwas rüder werde?) - das Einzige, was dabei tatsächlich noch "von mir" ist, wäre die Idee, wie man das machen könnte ... und die hatte ich sogar ganz konkret Dir schon vor Jahren im IPPF erklärt. Obendrein habe ich das auch noch in C umgesetzt, weil Du Dich selbst dazu nicht in der Lage gesehen hast - die entsprechenden IPPF-Beiträge hast Du allerdings auch selbst gelöscht.

---

Weiteres Lesen in Verbindung mit Freetz-NG tue ich mir aber auch nur an, wenn der Ton nicht so bleibt, wie er derzeit ist - und behaupte nicht wieder, Du würdest Deinerseits "nur reagieren", denn das hier: https://github.com/Freetz-NG/freetz-ng/discussions/328#discussioncomment-885675 ist schon ziemlich "eindeutig" in Bezug auf die - Deiner Ansicht nach "schlechten" - Flashprogramme/-Skripte, denn die Auswahl dabei ist nun nicht soo riesig.

Und ja ... ich bin hier absichtlich "etwas deutlicher" geworden - vielleicht hilft das ja, damit Du Dich künftig beherrschen lernst. Ich kann Dir auch versichern, daß ich meinerseits gar keinen Antrieb habe, hier zu schreiben, solange es nicht wieder mal erforderlich ist, meine Arbeit (sei es in Form von Tools oder von Beiträgen) gegen irgendwelche sinnfreien Anwürfe zu verteidigen. Zwar bringen solche Antworten, zu denen man dadurch quasi genötigt wird, insgesamt keinen Fortschritt - weder für Freetz-NG, noch für meine eigenen Absichten und Arbeiten. 

Trotzdem bin ich nicht bereit, irgendwelchen Unsinn dann einfach unwidersprochen stehen zu lassen - ich wäre insgesamt dafür, solche Diskussionen, die niemandem wirklich etwas bringen, zu vermeiden bzw. generell zu unterlassen. Ich wüßte auch nicht, daß ich meinerseits entsprechenden Streit vom Zaun gebrochen hätte - ich habe mich an die - selbst verordnete - Pause gehalten und hatte auch in dieser nicht den Eindruck, daß es hier "gesitteter" zugehen würde in der Zukunft.

Vielleicht solltest Du also wieder darüber nachdenken, mich ebenso wie andere einfach zu sperren (das gab es ja schon einmal in den "Anfangszeiten" (ca. die ersten 18 Monate), auch wenn es anderen sicherlich nicht aufgefallen ist - ich habe es trotzdem bemerkt) ... das erspart allen für die Zukunft weitere Diskussionen dieser Art und ich muß dann nicht mehr - für mich - entscheiden, ob ich nun helfen könnte und das auch will - das geht dann einfach nicht mehr, was jede weitere Überlegung in dieser Richtung überflüssig macht. Gleiches gilt dann für ähnliche Bemerkungen wie diejenigen, die mich hier nun erneut zu einer Reaktion veranlaßt haben (und das vermutlich ja auch sollten).

**Reply by fda77** [reply]
Ganz ehrlich, ich hab nicht die Zeit das alles zu lesen

**Reply by PeterPawn** [reply]
Du würdest es vermutlich auch nicht verstehen.

**Reply by fda77** [reply]
Dann merks dir fürs nächste mal und fass dich gleich kürzer

**Reply by PeterPawn** [reply]
Warum das denn? Bist Du jetzt der Einzige, FÜR DEN man hier etwas schreibt oder schreiben darf? Wenn Du es nicht lesen willst, laß es doch einfach sein - da brauchst Du Dir nicht noch ein Schild auf die Stirn zu pappen, damit jeder von Deiner Entscheidung (oder ist's am Ende Ignoranz) auch Kenntnis erlangen möge.

**Reply by fda77** [reply]
Prima, ich hab auch gar keine Lust auf alles antworten zu müssen. Das kannst du dann ja übernehmen
